packages/ldc/build.sh

```diff
@@ -0,0 +1,142 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/ldc-developers/ldc
+TERMUX_PKG_DESCRIPTION="D programming language compiler, built with LLVM"
+_PKG_MAJOR_VERSION=1.3
+TERMUX_PKG_VERSION=${_PKG_MAJOR_VERSION}.0-beta2
+TERMUX_PKG_SRCURL=https://github.com/ldc-developers/ldc/releases/download/v${TERMUX_PKG_VERSION}/ldc-${TERMUX_PKG_VERSION}-src.tar.gz
+TERMUX_PKG_SHA256=a6a13f356192d40649af7290820cf85127369f40d554c2fdd853dc098dce885f
+TERMUX_PKG_DEPENDS="clang"
+TERMUX_PKG_HOSTBUILD=true
+TERMUX_PKG_FORCE_CMAKE=yes
+#These CMake args are only used to configure a patched LLVM
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DLLVM_ENABLE_PIC=ON
+-DLLVM_BUILD_TOOLS=OFF
+-DLLVM_BUILD_UTILS=OFF
+-DPYTHON_EXECUTABLE=`which python`
+-DLLVM_TABLEGEN=$TERMUX_PKG_HOSTBUILD_DIR/bin/llvm-tblgen"
+TERMUX_PKG_KEEP_STATIC_LIBRARIES=true
+TERMUX_PKG_NO_DEVELSPLIT=yes
+TERMUX_PKG_MAINTAINER="Joakim @joakim-noah"
+
+termux_step_post_extract_package () {
+	# Just error out early on non-ARM
+	if [ $TERMUX_ARCH = "arm" ]; then
+		local LLVM_SRC_VERSION=3.9.1
+	else
+		termux_error_exit "Invalid arch: $TERMUX_ARCH"
+	fi
+	termux_download \
+		http://llvm.org/releases/${LLVM_SRC_VERSION}/llvm-${LLVM_SRC_VERSION}.src.tar.xz \
+		$TERMUX_PKG_CACHEDIR/llvm-${LLVM_SRC_VERSION}.src.tar.xz \
+		1fd90354b9cf19232e8f168faf2220e79be555df3aa743242700879e8fd329ee
+
+	tar xf $TERMUX_PKG_CACHEDIR/llvm-${LLVM_SRC_VERSION}.src.tar.xz
+	mv llvm-${LLVM_SRC_VERSION}.src llvm
+
+	DMD_COMPILER_VERSION=2.074.1
+	termux_download \
+		http://downloads.dlang.org/releases/2.x/${DMD_COMPILER_VERSION}/dmd.${DMD_COMPILER_VERSION}.linux.tar.xz \
+		$TERMUX_PKG_CACHEDIR/dmd.${DMD_COMPILER_VERSION}.linux.tar.xz \
+		e48783bd91d77bfdcd702bd268c5ac5d322975dd4b3ad68831babd74509d2ce9
+
+	sed "s#\@TERMUX_C_COMPILER\@#$TERMUX_STANDALONE_TOOLCHAIN/bin/$TERMUX_HOST_PLATFORM-clang#" \
+		$TERMUX_PKG_BUILDER_DIR/ldc-config-stdlib.patch.beforehostbuild.in > \
+		$TERMUX_PKG_BUILDER_DIR/ldc-config-stdlib.patch.beforehostbuild
+	sed -i "s#\@TERMUX_C_FLAGS\@#-march=armv7-a -mfpu=neon -mfloat-abi=softfp -mthumb -Os -I$TERMUX_PREFIX/include#" \
+		$TERMUX_PKG_BUILDER_DIR/ldc-config-stdlib.patch.beforehostbuild
```

Comment:
Pull in some NDK flags for the host build, to cross-compile the D stdlib for ARM.